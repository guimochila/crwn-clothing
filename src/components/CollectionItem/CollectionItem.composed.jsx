import React from 'react';
import { Mutation } from 'react-apollo';

import CollectionItem from './CollectionItem';
import { ADD_ITEM_TO_CART } from '../../graphql/queries';

const CollectionItemComposed = props => (
  <Mutation mutation={ADD_ITEM_TO_CART} variables={{ item: props.item }}>
    {addItemToCart => (
      <CollectionItem {...props} addItem={item => addItemToCart(item)} />
    )}
  </Mutation>
);

export default CollectionItemComposed;
