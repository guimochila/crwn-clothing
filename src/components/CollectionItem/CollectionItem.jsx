import React from 'react';
import {
  CollectionItemContainer,
  CollectionFooterContainer,
  AddButton,
  BackgroundImage,
  NameContainer,
  PriceContainer,
} from './CollectionItem.styles';

function CollectionItem({ item, addItem }) {
  const { name, price, imageUrl } = item;

  return (
    <CollectionItemContainer>
      <BackgroundImage className="image" imageUrl={imageUrl} />
      <CollectionFooterContainer>
        <NameContainer>{name}</NameContainer>
        <PriceContainer>{price}</PriceContainer>
      </CollectionFooterContainer>
      <AddButton inverted onClick={() => addItem(item)}>
        Add to cart
      </AddButton>
    </CollectionItemContainer>
  );
}

export default CollectionItem;
