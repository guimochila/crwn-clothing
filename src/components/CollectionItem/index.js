import CollectionItemComposed from './CollectionItem.composed';

export default CollectionItemComposed;
