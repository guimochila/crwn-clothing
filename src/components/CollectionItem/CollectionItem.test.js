import React from 'react';
import { shallow } from 'enzyme';

import { CollectionItem } from './CollectionItem';

describe('CollectionItem component', () => {
  let wrapper;
  const mockProps = {
    item: {
      name: 'Hat',
      price: 20,
      imageUrl: 'https://imgsrc.com/hat',
    },
    addItem: jest.fn(),
  };

  beforeEach(() => {
    wrapper = shallow(<CollectionItem {...mockProps} />);
  });

  test('should render CollectionItem', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should call addItem with item when AddButton is clicked', () => {
    wrapper.find('AddButton').simulate('click');
    expect(mockProps.addItem).toHaveBeenCalledWith(mockProps.item);
  });

  test('should render imageUrl as a prop on BackgroundImage', () => {
    expect(wrapper.find('BackgroundImage').prop('imageUrl')).toBe(
      mockProps.item.imageUrl,
    );
  });

  test('should render name prop in NameContainer', () => {
    expect(wrapper.find('NameContainer').text()).toBe(mockProps.item.name);
  });

  test('should render price prop in PriceContainer', () => {
    const price = parseInt(wrapper.find('PriceContainer').text());
    expect(price).toBe(mockProps.item.price);
  });
});
