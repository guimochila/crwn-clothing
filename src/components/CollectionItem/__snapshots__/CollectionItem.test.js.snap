// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionItem component should render CollectionItem 1`] = `
<styled.div>
  <BackgroundImage
    className="image"
    imageUrl="https://imgsrc.com/hat"
  />
  <styled.div>
    <NameContainer>
      Hat
    </NameContainer>
    <PriceContainer>
      20
    </PriceContainer>
  </styled.div>
  <AddButton
    inverted={true}
    onClick={[Function]}
  >
    Add to cart
  </AddButton>
</styled.div>
`;
